zetaclient/orchestrator/orchestrator.go

```diff
@@ -100,20 +103,20 @@ func NewOrchestrator(
 }
 
 // MonitorCore starts the orchestrator for CCTXs
-func (oc *Orchestrator) MonitorCore(appContext *context.AppContext) error {
+func (oc *Orchestrator) MonitorCore(ctx context.Context) error {
 	signerAddress, err := oc.zetacoreClient.GetKeys().GetAddress()
 	if err != nil {
 		return fmt.Errorf("failed to get signer address: %w", err)
 	}
 	oc.logger.Std.Info().Msgf("Starting orchestrator for signer: %s", signerAddress)
 
 	// start cctx scheduler
-	go oc.StartCctxScheduler(appContext)
+	bg.Work(ctx, oc.StartCctxScheduler, bg.WithName("StartCctxScheduler"), bg.WithLogger(oc.logger.Std))
 
 	// watch for upgrade plan from zetacore
 	go func() {
 		// wait for upgrade plan signal to arrive
-		oc.zetacoreClient.Pause()
+		oc.zetacoreClient.Stop()
```

Comment:
Is this a real stop? The previous `Pause()` is waiting for upgrade plan to arrive its height?

Reply:
This is a good question. Can you elaborate on initial `Pause()` use-case? Because the log says
```
Msgf("Active upgrade plan detected and upgrade height reached: %s at height %d; ZetaClient is stopped;"+
				"please kill this process, replace zetaclientd binary with upgraded version, and restart zetaclientd", plan.Name, plan.Height)
```


https://github.com/zeta-chain/node/blob/develop/zetaclient/zetacore/client.go#L206

Reply:
UPD: I followed the logic in `develop` branch. I found `Pause()` method confusing as it doesn't "pause" the client but blocks the current go-routine until someone pushes `c.pause <- struct{}{}`. 

I'll refactor code accordingly, thanks for spotting this!